lib/internal/modules/esm/resolve.js

```diff
@@ -1087,6 +1063,40 @@ function defaultResolve(specifier, context = {}) {
     // Ignore exception
   }
 
+  // Import maps are processed first so that policies can still apply
+  // to the resolved specifier and also so the other resolution logic
+  // happens on the mapped specifier
+  if (importMap) {
+    // Intentionally mutating here as we don't think it is a problem
+    // since it is a primitive
+    specifier = importMap.resolve(parsed || specifier, parsedParentURL);
```

Comment:
I _think_ you might need a nullish or instead of a falsy or:

```suggestion
    specifier = importMap.resolve(parsed ?? specifier, parsedParentURL);
```

Reply:
Not sure it is strictly necessary but I am fine changing it. Is there a case where `new URL` could truthy but also not a correctly parsed URL instance?

Reply:
I am going to mark this resolve in here, but will be opening the real PR soon so we can re-visit this if we need to there.